src/raftstore/store/peer.rs

```diff
@@ -545,6 +550,11 @@ impl Peer {
         // Always update read delegate's region to avoid stale region info after a follower
         // becomeing a leader.
         self.maybe_update_read_progress(progress);
+
+        if !self.pending_remove {
+            self.coprocessor_host
```

Comment:
Why can it be skipped?

Reply:
And for every proposal we need to emit an event? Maybe there is too many.

Reply:
@overvenus I just found that `maybe_update_read_progress` checked the `pending_remove` like this. I think in this condition it's useless to trigger the observer here, so I checked it too.

Reply:
@hicqu Did you mean that the `set_region` function is actually invoked on every proposal? I just want to listen all changes of the `meta::Region`s.